packages/compiler-sfc/src/compileScript.ts

```diff
@@ -572,6 +572,36 @@ export function compileScript(
     return `\n  props: ${propsDecls} as unknown as undefined,`
   }
 
+  function genSetupPropsType(
+    props: Record<string, PropTypeData>,
+    propsType: string
+  ) {
+    const keys = Object.keys(props)
+    if (!keys.length) {
+      return ``
+    }
+    const hasStaticDefaults =
```

Comment:
Maybe this should be refactored. eg: a method `hasStaticDefaults()` should be extracted because the same logic already exists in `genRuntimeProps`.

Reply:
@edison1105 changed,thanks for your suggest